config/routes.rb

```diff
@@ -8,75 +16,58 @@
   get '/search', to: 'search#find'
 
   # get w id -> show action
   get '/annotations/:anno_root/:id(.:format)', to: 'annotations#show',
-    constraints: lambda { |request|
-      anno_root = request.path_parameters[:anno_root]
-      anno_root !~ /^search$/ && anno_root !~ /^new$/
-    }
+    constraints: lambda { |request| annotations_container_exclusions request }
   get '/:anno_root/:id(.:format)', to: 'annotations#show',
-    constraints: lambda { |request|
-      anno_root = request.path_parameters[:anno_root]
-      id = request.path_parameters[:id]
-      anno_root !~ /^annotations$/ && anno_root !~ /^search$/ && anno_root !~ /^new$/ && id !~ /^search$/ && id !~ /^new$/
-    }
+    constraints: lambda { |request| anno_container_exclusions request }
 
   # get no id -> index action
   get '/annotations/:anno_root', to: 'annotations#index',
-    constraints: lambda { |request|
-      anno_root = request.path_parameters[:anno_root]
-      anno_root !~ /^search$/ && anno_root !~ /^new$/
-    }
+    constraints: lambda { |request| annotations_container_exclusions request }
   get '/:anno_root', to: 'annotations#index',
-    constraints: lambda { |request|
-      anno_root = request.path_parameters[:anno_root]
-      anno_root !~ /^annotations$/ && anno_root !~ /^search$/ && anno_root !~ /^new$/
-    }
+    constraints: lambda { |request| anno_container_exclusions request }
 
   # post -> create action
   post '/annotations/:anno_root', to: 'annotations#create',
-    constraints: lambda { |request|
-      anno_root = request.path_parameters[:anno_root]
-      id = request.path_parameters[:id]
-      anno_root !~ /^search$/ && anno_root !~ /^new$/ && id !~ /^search$/ && id !~ /^new$/
-    }
+    constraints: lambda { |request| annotations_container_exclusions request }
   post '/:anno_root', to: 'annotations#create',
-    constraints: lambda { |request|
-      anno_root = request.path_parameters[:anno_root]
-      anno_root !~ /^annotations$/ && anno_root !~ /^search$/ && anno_root !~ /^new$/
-    }
+    constraints: lambda { |request| anno_container_exclusions request }
 
   # delete -> destroy action
   delete '/annotations/:anno_root/:id(.:format)', to: 'annotations#destroy',
-    constraints: lambda { |request|
-      anno_root = request.path_parameters[:anno_root]
-      id = request.path_parameters[:id]
-      anno_root !~ /^search$/ && anno_root !~ /^new$/ && id !~ /^search$/ && id !~ /^new$/
-    }
+    constraints: lambda { |request| annotations_container_exclusions request }
   delete '/:anno_root/:id(.:format)', to: 'annotations#destroy',
-    constraints: lambda { |request|
-      anno_root = request.path_parameters[:anno_root]
-      anno_root !~ /^annotations$/ && anno_root !~ /^search$/ && anno_root !~ /^new$/
-    }
-
+    constraints: lambda { |request| anno_container_exclusions request }
 
   get '/annotations', to: 'search#find'
   root to: 'search#find'
 
 end
+
+
```

Comment:
perhaps it would be more clear above if id_exclusions was not part of container_exclusions.   I think it's fine to exclude "annotations" as an auth_root value for everything.